proof/refine/ARM/Reply_R.thy

```diff
@@ -439,23 +440,63 @@ lemma isHead_to_head:
   apply (case_tac y; clarsimp)
   done
 
-(* FIXME RT: remove. proved by MikiT *)
-lemma sym_refs_replyNext_replyPrev_proj:
+(* sym_heap *)
+
+(* RT FIXME: move? *)
+lemma sym_heap_list_rev:
+  "\<lbrakk>sym_heap hp hp'; heap_list hp (Some p) ps\<rbrakk>
+  \<Longrightarrow> heap_path hp' (hd_opt (rev ps)) (butlast (rev ps)) (Some p)
+      \<and> (hd_opt |> hp) (rev ps) = None"
```

Comment:
I think that you can show `heap_list hp (Some p) ps ⟹ ps ≠ []`, which would let you use `hp (last ps)` instead of the complications with `hd_opt` and `rev`.